pkgs/applications/emulators/cdemu/analyzer.nix

```diff
@@ -1,20 +1,14 @@
-{ callPackage, makeWrapper, gobject-introspection, cmake
+{ callPackage, makeWrapper, gobject-introspection, cmake, wrapGAppsHook3
 , python3Packages, gtk3, glib, libxml2, gnuplot, gnome, gdk-pixbuf, librsvg, intltool, libmirage }:
-let pkg = import ./base.nix {
-  version = "3.2.5";
+callPackage ./base.nix {
+  version = "3.2.6";
   pname = "image-analyzer";
-  pkgSha256 = "00906lky0z1m0bdqnjmzxgcb19dzvljhddhh42lixyr53sjp94cc";
-};
-in callPackage pkg {
+  sha256 = "sha256-7I8RUgd+k3cEzskJGbziv1f0/eo5QQXn62wGh/Y5ozc=";
   buildInputs = [ glib gtk3 libxml2 gnuplot libmirage gnome.adwaita-icon-theme gdk-pixbuf librsvg
-                  python3Packages.python python3Packages.pygobject3 python3Packages.matplotlib ];
-  drvParams = {
-    nativeBuildInputs = [ gobject-introspection cmake makeWrapper intltool ];
-    postFixup = ''
-      wrapProgram $out/bin/image-analyzer \
-        --set PYTHONPATH "$PYTHONPATH" \
-        --set GI_TYPELIB_PATH "$GI_TYPELIB_PATH" \
-        --prefix XDG_DATA_DIRS : "$out/share:$XDG_ICON_DIRS:$GSETTINGS_SCHEMAS_PATH"
-    '';
-  };
+                  python3Packages.pygobject3 python3Packages.matplotlib ];
+  nativeBuildInputs = [ gobject-introspection cmake wrapGAppsHook3 intltool ];
+  postFixup = ''
+    wrapProgram $out/bin/image-analyzer \
```

Comment:
Take care to avoid [double wrapping](https://nixos.org/manual/nixpkgs/unstable/#ssec-gnome-common-issues-double-wrapped).

Reply:
I'm very aware, and that was also why I asked in matrix about it. I'll try your suggestion of using `buildPythonApplication` with `pyproject = false`

Reply:
Turns out, that `buildPythonApplication` with `pyproject = false` has the same behavior as `mkDerivation` with `wrapPython`.

- we've still got double-wrapping from `wrapGAppsHook`
- the resulting binary still doesn't find its path:

I think I'll merge this as is for now. We can improve in a separate PR. Maybe that's also a bug with wrapPython? Is that supposed to pick up buildInputs?

```nix
{ fetchurl, python3Packages, cmake, pkg-config, intltool, wrapGAppsNoGuiHook }:
python3Packages.buildPythonApplication rec {
  version = "3.2.5";
  pname = "cdemu-client";
  src = fetchurl {
    url = "mirror://sourceforge/cdemu/${pname}-${version}.tar.xz";
    hash = "sha256-py2F61v8vO0BCM18GCflAiD48deZjbMM6wqoCDZsOd8=";
  };
  pyproject = false;
  nativeBuildInputs = [ cmake pkg-config intltool wrapGAppsNoGuiHook ];
  buildInputs = with python3Packages; [ dbus-python pygobject3 ];
}
```

```console
± ./result/bin/cdemu status
Traceback (most recent call last):
  File "/nix/store/r0rkqb7zv750k4jckpj4cys3nl07jqql-cdemu-client-3.2.5/bin/..cdemu-wrapped-wrapped", line 33, in <module>
    import gi
ModuleNotFoundError: No module named 'gi'
```

Reply:
I fixed the double wrapping, but still explicitly set PYTHONPATH. I think I'll leave this in for now. Most pressing feedback is fixed and we can always revisit

Reply:
Partially my bad: I managed to fix this with the info you provided @jtojnar 

```sh
makeWrapperArgs+=("''${gappsWrapperArgs[@]}")
```

However: When going that route, we suddenly only have `dconf` and `gobject-introspection` on the `GI_TYPELIB_PATH`, which makes the gtk3 import fail:

```console
ValueError: Namespace Gtk not available
```

Reply:
> Is that supposed to pick up buildInputs?

No. Python modules are only picked from `propagatedBuildInputs` and `pythonPath`.

Reply:
See [the manual](https://nixos.org/manual/nixpkgs/unstable/#ssec-gnome-common-issues-double-wrapped) for how to resolve the double wrapping.

Reply:
I did both. It already works. But I change it to `buildPythonApplication` together with `makeWrapperArgs+=("''${gappsWrapperArgs[@]}")`, then it suddenly doesn't any more.

Reply:
@jtojnar am I doing something wrong here? https://github.com/bendlas/nixpkgs/commit/e4ab3dfcbbc298993c9e59fca6d3c57947237d4b

```console
± $(nix-build . -A image-analyzer --no-out-link)/bin/image-analyzer
Traceback (most recent call last):
  File "/nix/store/lgnv7ay3hi94rzwxvxj1c7k0xi7swy3x-image-analyzer-3.2.6/bin/.image-analyzer-wrapped", line 26, in <module>
    gi.require_version('Gtk', '3.0')
  File "/nix/store/9njwgmbs7jlpsrnnr731k2vv5xqghqr4-python3.11-pygobject-3.48.2/lib/python3.11/site-packages/gi/__init__.py", line 122, in require_version
    raise ValueError('Namespace %s not available' % namespace)
ValueError: Namespace Gtk not available
```

As I said, `GI_TYPELIB_PATH` is missing a lot of stuff, compared to the `stdenv.mkDerivation` variant, that's currently in master

```console
± less $(nix-build . -A image-analyzer --no-out-link)/bin/image-analyzer
...
# The C-code for this binary wrapper has been generated using the following command:
makeCWrapper '/nix/store/lgnv7ay3hi94rzwxvxj1c7k0xi7swy3x-image-analyzer-3.2.6/bin/.image-analyzer-wrapped' \
   ...
    --prefix 'GI_TYPELIB_PATH' ':' '/nix/store/bcms6vbac6jyffwyj8y630brpy1il4ks-glib-2.80.0/lib/girepository-1.0:/nix/store/ys2ajfa1bbx0a3axb722na18ymw8h9jy-gobject-introspection-1.80.1/lib/girepository-1.0'
...
```

Reply:
You are missing `gobject-introspection.setupHook` from `nativeBuildInputs`.

Reply:
that works. let's continue here https://github.com/NixOS/nixpkgs/pull/311959